R/modify-model-field.R

```diff
@@ -178,6 +185,72 @@ remove_tags <- function(.mod, .tags) {
     .append = FALSE,
     .remove = TRUE,
     .char_value = TRUE,
+    .bool_value = FALSE
+  )
+}
+
+
+
+#' @name add_star
+#' @title Star a model
+#'
+#' @description Adds star  on a model object and corresponding YAML.
+#' Tags can be modified at any time (i.e. before _or_ after a model is submitted).
+#'
+#' @details
+#' The `starred` fields on a `bbi_{.model_type}_model` object contains a boolean variable if model is starred
+#'
+#'
+#' @return The modified `bbi_{.model_type}_model` object
+#'
+#' @seealso [run_log()] [collapse_to_string()] [modify_notes()] [modify_based_on()] [modify_description()] [modify_bbi_args()]
+#' NULL
+#' @inheritParams modify_model_field
+#' @param .starred boolean variable to indicate significance
+#' @export
+add_star <- function(.mod, .starred) {
+
+  modify_model_field(
+    .mod = .mod,
+    .field = YAML_STAR,
+    .value = .starred,
+    .append = FALSE,
+    .char_value = FALSE,
+    .bool_value = TRUE
+  )
+}
+
+#' @title replace_star
+#' @param .starred boolean variable to indicate significance
+#' @param .mod list object of model data
+#' @describeIn Add or remove star on a model object and corresponding YAML.
+#' Tags can be modified at any time (i.e. before _or_ after a model is submitted).
+#' @export
+replace_star <- function(.mod, .starred) {
```

Comment:
Like Seth, I don't see the purpose of having this function.

Reply:
I created this function to follow the conventions. I was mostly looking around line 68 at this code chunk. 
` if (!is.null(.description)) .mod <- replace_description(.mod, .description)
  if (!is.null(.tags))        .mod <- replace_all_tags(.mod, .tags)
  if (!is.null(.bbi_args))    .mod <- replace_all_bbi_args(.mod, .bbi_args)
  if (!is.null(.based_on))    .mod <- replace_all_based_on(.mod, .based_on)
  if (!is.null(.starred))     .mod <- replace_star(.mod, .starred)
  return(.mod)`
Would it be alright to use like add_star instead?

Reply:
I see, so here:
https://github.com/metrumresearchgroup/bbr/blob/74baeb15d44eab79fdd110fb81626fbea9212b50/R/new-model.R#L37-L72

> Would it be alright to use like add_star instead?

Hmm, so my understanding is that `.starred = NULL` means "leave as is".  So, yeah, one option would be: if `.starred = TRUE`, call `add_star(.mod)` and if `.starred = FALSE` call `remove_star(.mod)`.  However, my vote would be to call `modify_model_field` directly, basically inlining the current body of `replace_star`. 

I don't think the details matter too much, though, as long as `replace_star` isn't *exported* (because the above is an internal implementation detail, and I can't think of when a user would want to call `replace_start`).